core/tx_pool.go

```diff
@@ -1176,6 +1226,33 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 			reinject = types.TxDifference(discarded, included)
 		}
 	}
+
+	// for AA we need to re-validate accounts that were included
+	// split into separate function
+	// This introduces some basic race conditions, so this needs to be refactored
+	for _, tx := range included {
+		var account = tx.To()
+		if account != nil && tx.IsAA() {
+			if pool.Has(tx.Hash()) {
+				pool.removeTx(tx.Hash(), true)
```

Comment:
This only occurs if that account was touched in a new block that comes through. This will make re-validation occur. We remove it from the pool then add it again through the queue which gets it validated.